x/qgb/versions.go

```diff
@@ -0,0 +1,17 @@
+package qgb
+
+import (
+	v1 "github.com/celestiaorg/celestia-app/x/qgb/v1"
+	"github.com/celestiaorg/celestia-app/x/qgb/v1beta1"
+)
+
+func GetSignificantPowerDiffThreshold(appVersion uint64) float64 {
+	switch appVersion {
+	case 0:
+		return v1beta1.SignificantPowerDifferenceThreshold
+	case 1:
+		return v1.SignificantPowerDifferenceThreshold
+	default:
+		return v1.SignificantPowerDifferenceThreshold
```

Comment:
[question] should this panic with something like "unsupported appVersion"

Reply:
this mechanism could be refactored to be clearer/simpler, we just need to make sure that we use v1 for all app versions greater than zero

Reply:
> we just need to make sure that we use v1 for all app versions greater than zero

I think a v1 validator would use an incorrect significant power different threshold for a block produced with app version 2. Instead of allowing that calculation to take place and produce an incorrect result, why not error / panic here?

Reply:
I guess this isn't blocking b/c it's future proofing but it does mean that a v0 app would handle a v2 block incorrectly rather than panicing.

Reply:
Say we merge this PR as-is and cut a v0.13.0 release. celestia-app v0.13.0:

appVersion 0 => v0 SignificantPowerDifferenceThreshold
appVersion 1 => v1 SignificantPowerDifferenceThreshold
appVersion 2 => v1 SignificantPowerDifferenceThreshold

At some point in the future we introduce a new significant power difference threshold that should apply for appVersion 2. We cut a v1.0.0 release. celestia-app v1.0.0:

appVersion 0 => v0 SignificantPowerDifferenceThreshold
appVersion 1 => v1 SignificantPowerDifferenceThreshold
appVersion 2 => v2 SignificantPowerDifferenceThreshold

Then most of the network validators upgrades to the v1.0.0 binary but a few stragglers remain on the v0.13.0 binary. When a block height is reached where appVersion 2 should be used, the un-upgraded validators incorrectly use the v1 SignificantPowerDifferenceThreshold instead of `panic`. Unless there is a code block that get executed before this where they try to validate a v2 appVersion block and panic immediately but if that's the case, I think we should do the same here for clarity.

Reply:
1) all un-upgraded nodes will reject blocks that use a different appversion then what they are using before they even execute any transactions.

2) if they panic, then for each upgrade, we have to change _every_ if statement switch to support the new version, but that doesn't actually give us any benefit due to point 1. 

if we need to change the constant after some number of app versions then we add
appVersion N => v2 SignificantPowerDifferenceThreshold

Reply:
> 1. all un-upgraded nodes will reject blocks that use a different appversion then what they are using before they even execute any transactions.

Good point. Given this, I don't think it's necessary to do the `panic` thing I was describing.